stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsActivity.kt

```diff
@@ -138,15 +147,33 @@ internal class PaymentOptionsActivity : BasePaymentSheetActivity<PaymentOptionRe
                 }
             }
         }
+    }
 
-        viewModel.userSelection.observe(this) { paymentSelection ->
-            if (paymentSelection != null) {
-                onActionCompleted(paymentSelection)
+    @VisibleForTesting
+    private val viewStateObserver: (ViewState.PaymentOptions?) -> Unit
+        get() = { viewState ->
+            lifecycleScope.launch {
+                val addButton = viewBinding.addButton
+                addButton.updateState(
+                    when (viewState) {
+                        null -> null
+                        is ViewState.PaymentOptions.Ready -> PrimaryButton.State.Ready(addButtonLabel)
+                        is ViewState.PaymentOptions.Confirming -> PrimaryButton.State.Confirming
+                        is ViewState.PaymentOptions.Completed -> PrimaryButton.State.Completed
```

Comment:
Does this handle the cases where we don't want to animate the "Add" button?

Reply:
I was going to try to descope that in an effort to shrink the size of my PRs.  Right now this will animate the Add button every time regardless of if it is a save operation or not.